apps/src/studio/studio.js

```diff
@@ -1142,11 +1144,13 @@ Studio.clearEventHandlersKillTickLoop = function() {
     });
   }
   Studio.eventHandlers = [];
-  if (Studio.intervalId) {
-    window.clearInterval(Studio.intervalId);
-  }
+  if (Studio.perExecutionTimeouts) {
+    Studio.perExecutionTimeouts.forEach(function (timeout) {
+      clearInterval(timeout);
+    });
+   }
```

Comment:
How you have it is okay, it just adds an extra level of indentation. If it were me I would probably try to clean up both perExecutionTimeouts and eventHandlers, but it's fine if you leave it as is.